tubular/asgard.py

```
@@ -176,6 +178,36 @@ def new_asg(cluster, ami_id):
 
     return new_asg
 
+def is_asg_pending_delete(asg):
+    """
+    Checks status of an ASG, specifically if it is pending deletion.
+
+    Argument:
+        asg(str): ASG whose status should be checked.
+
+    Returns:
+        True if the asg is in the "pending delete" status, else return False.
+    """
+
+    url = ASG_INFO_URL.format(asg)
+    LOG.debug("URL: {}".format(url))
+    response = requests.get(url, params=ASGARD_API_TOKEN, timeout=REQUESTS_TIMEOUT)
+
+    if response.status_code == 404:
+        raise BackendError('Autoscale group {} does not exist'.format(asg))
```

@feanil what is your suggestion here? If an ASG doesn't exist, it doesn't exist. We could raise an exception of a different type in this situation, but ultimately the system ends up reacting the same way: Tubular uses this call to see if an ASG is pending delete, and if it is, it halts it's current operation. 

Lets think about it this way for a moment as well. Say the caller of this function passes a bad ASG name. The ASG doesn't exist and we return that this ASG is not pending delete. Their code then goes on from here to delete or disable other ASGs or do whatever it does, while in fact the ASG they had intended to query is actually in the process or a delete operation. The caller is essentially acting on false information.